### ISSUE TITLE
Fix / document setup process

### ISSUE THREAD

**Issue by homersimpsons** [RustPython/rustpython.github.io#65]
## Context

I wanted to solve the issue #66 where images are not displayed for https://rustpython.github.io/benchmarks

## Issue

I was not able to start Jelykill in [GitHub codespaces](https://github.com/features/codespaces).

## Reproduction

Steps to start jekyll in [GitHub codespaces](https://github.com/features/codespaces):
1. Open a GitHub codespace for this project
<details><summary>2. Run <code>bundle install</code></summary>

```
Bundler 2.4.10 is running, but your lockfile was generated with 2.1.4. Installing Bundler 2.1.4 and restarting using that version.
Fetching gem metadata from https://rubygems.org/.
Fetching bundler 2.1.4
Installing bundler 2.1.4
Calling `DidYouMean::SPELL_CHECKERS.merge!(error_name => spell_checker)' has been deprecated. Please call `DidYouMean.correct_error(error_name, spell_checker)' instead.
Fetching gem metadata from https://rubygems.org/...........
Fetching public_suffix 4.0.6
Installing public_suffix 4.0.6
Fetching addressable 2.8.0
Installing addressable 2.8.0
Using bundler 2.1.4
Using colorator 1.1.0
Fetching concurrent-ruby 1.1.8
Installing concurrent-ruby 1.1.8
Using eventmachine 1.2.7
Fetching http_parser.rb 0.6.0
Installing http_parser.rb 0.6.0 with native extensions
Fetching em-websocket 0.5.2
Installing em-websocket 0.5.2
Fetching ffi 1.15.0
Installing ffi 1.15.0 with native extensions
Using forwardable-extended 2.6.0
Fetching i18n 1.8.9
Installing i18n 1.8.9
Fetching sassc 2.4.0
Installing sassc 2.4.0 with native extensions
Fetching jekyll-sass-converter 2.1.0
Installing jekyll-sass-converter 2.1.0
Fetching rb-fsevent 0.10.4
Installing rb-fsevent 0.10.4
Using rb-inotify 0.10.1
Fetching listen 3.5.0
Installing listen 3.5.0
Using jekyll-watch 2.2.1
Using rexml 3.2.5
Fetching kramdown 2.3.1
Installing kramdown 2.3.1
Using kramdown-parser-gfm 1.1.0
Fetching liquid 4.0.3
Installing liquid 4.0.3
Using mercenary 0.4.0
Using pathutil 0.16.2
Fetching rouge 3.26.0
Installing rouge 3.26.0
Using safe_yaml 1.0.5
Fetching unicode-display_width 1.7.0
Installing unicode-display_width 1.7.0
Fetching terminal-table 2.0.0
Installing terminal-table 2.0.0
Fetching jekyll 4.2.0
Installing jekyll 4.2.0
Fetching jekyll-feed 0.15.1
Installing jekyll-feed 0.15.1
Fetching jekyll-redirect-from 0.16.0
Installing jekyll-redirect-from 0.16.0
Fetching jekyll-seo-tag 2.7.1
Installing jekyll-seo-tag 2.7.1
Fetching minima 2.5.1
Installing minima 2.5.1
Bundle complete! 7 Gemfile dependencies, 32 gems now installed.
Use `bundle info [gemname]` to see where a bundled gem is installed.
```
</details>
<details><summary>3. Run <code>bundle exec jekyll serve</code> (failing)</summary>

```
Calling `DidYouMean::SPELL_CHECKERS.merge!(error_name => spell_checker)' has been deprecated. Please call `DidYouMean.correct_error(error_name, spell_checker)' instead.
Calling `DidYouMean::SPELL_CHECKERS.merge!(error_name => spell_checker)' has been deprecated. Please call `DidYouMean.correct_error(error_name, spell_checker)' instead.
Configuration file: /workspaces/rustpython.github.io/_config.yml
            Source: /workspaces/rustpython.github.io
       Destination: /workspaces/rustpython.github.io/_site
 Incremental build: disabled. Enable with --incremental
      Generating... 
       Jekyll Feed: Generating feed for posts
  Liquid Exception: undefined method `tainted?' for "%b %-d, %Y":String in /workspaces/rustpython.github.io/_layouts/post.html
                    ------------------------------------------------
      Jekyll 4.2.0   Please append `--trace` to the `serve` command 
                     for any additional information or backtrace. 
                    ------------------------------------------------
/usr/local/rvm/gems/ruby-3.2.2/gems/liquid-4.0.3/lib/liquid/variable.rb:124:in `taint_check': undefined method `tainted?' for "%b %-d, %Y":String (NoMethodError)

      return unless obj.tainted?
                       ^^^^^^^^^
        from /usr/local/rvm/gems/ruby-3.2.2/gems/liquid-4.0.3/lib/liquid/variable.rb:89:in `render'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/liquid-4.0.3/lib/liquid/tags/assign.rb:26:in `render'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/liquid-4.0.3/lib/liquid/block_body.rb:103:in `render_node_to_output'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/liquid-4.0.3/lib/liquid/block_body.rb:91:in `render'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/liquid-4.0.3/lib/liquid/template.rb:208:in `block in render'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/liquid-4.0.3/lib/liquid/template.rb:242:in `with_profiling'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/liquid-4.0.3/lib/liquid/template.rb:207:in `render'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/liquid-4.0.3/lib/liquid/template.rb:220:in `render!'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/jekyll-4.2.0/lib/jekyll/liquid_renderer/file.rb:39:in `block (3 levels) in render!'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/jekyll-4.2.0/lib/jekyll/liquid_renderer/file.rb:59:in `measure_counts'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/jekyll-4.2.0/lib/jekyll/liquid_renderer/file.rb:38:in `block (2 levels) in render!'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/jekyll-4.2.0/lib/jekyll/liquid_renderer/file.rb:63:in `measure_bytes'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/jekyll-4.2.0/lib/jekyll/liquid_renderer/file.rb:37:in `block in render!'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/jekyll-4.2.0/lib/jekyll/liquid_renderer/file.rb:70:in `measure_time'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/jekyll-4.2.0/lib/jekyll/liquid_renderer/file.rb:36:in `render!'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/jekyll-4.2.0/lib/jekyll/renderer.rb:131:in `render_liquid'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/jekyll-4.2.0/lib/jekyll/renderer.rb:194:in `render_layout'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/jekyll-4.2.0/lib/jekyll/renderer.rb:163:in `place_in_layouts'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/jekyll-4.2.0/lib/jekyll/renderer.rb:93:in `render_document'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/jekyll-4.2.0/lib/jekyll/renderer.rb:63:in `run'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/jekyll-4.2.0/lib/jekyll/site.rb:547:in `render_regenerated'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/jekyll-4.2.0/lib/jekyll/site.rb:532:in `block (2 levels) in render_docs'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/jekyll-4.2.0/lib/jekyll/site.rb:531:in `each'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/jekyll-4.2.0/lib/jekyll/site.rb:531:in `block in render_docs'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/jekyll-4.2.0/lib/jekyll/site.rb:530:in `each_value'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/jekyll-4.2.0/lib/jekyll/site.rb:530:in `render_docs'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/jekyll-4.2.0/lib/jekyll/site.rb:210:in `render'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/jekyll-4.2.0/lib/jekyll/site.rb:80:in `process'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/jekyll-4.2.0/lib/jekyll/command.rb:28:in `process_site'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/jekyll-4.2.0/lib/jekyll/commands/build.rb:65:in `build'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/jekyll-4.2.0/lib/jekyll/commands/build.rb:36:in `process'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/jekyll-4.2.0/lib/jekyll/command.rb:91:in `block in process_with_graceful_fail'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/jekyll-4.2.0/lib/jekyll/command.rb:91:in `each'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/jekyll-4.2.0/lib/jekyll/command.rb:91:in `process_with_graceful_fail'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/jekyll-4.2.0/lib/jekyll/commands/serve.rb:86:in `block (2 levels) in init_with_program'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/mercenary-0.4.0/lib/mercenary/command.rb:221:in `block in execute'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/mercenary-0.4.0/lib/mercenary/command.rb:221:in `each'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/mercenary-0.4.0/lib/mercenary/command.rb:221:in `execute'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/mercenary-0.4.0/lib/mercenary/program.rb:44:in `go'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/mercenary-0.4.0/lib/mercenary.rb:21:in `program'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/jekyll-4.2.0/exe/jekyll:15:in `<top (required)>'
        from /usr/local/rvm/gems/ruby-3.2.2/bin/jekyll:25:in `load'
        from /usr/local/rvm/gems/ruby-3.2.2/bin/jekyll:25:in `<main>'
        from /usr/local/rvm/gems/ruby-3.2.2/bin/ruby_executable_hooks:22:in `eval'
        from /usr/local/rvm/gems/ruby-3.2.2/bin/ruby_executable_hooks:22:in `<main>'
```
</details>

It looks like the above issue is fixed in: https://github.com/Shopify/liquid/issues/1625

<details><summary>But the <code>bundle exec jekyll serve</code> command still is failing</summary>

```
Calling `DidYouMean::SPELL_CHECKERS.merge!(error_name => spell_checker)' has been deprecated. Please call `DidYouMean.correct_error(error_name, spell_checker)' instead.
Calling `DidYouMean::SPELL_CHECKERS.merge!(error_name => spell_checker)' has been deprecated. Please call `DidYouMean.correct_error(error_name, spell_checker)' instead.
Configuration file: /workspaces/rustpython.github.io/_config.yml
            Source: /workspaces/rustpython.github.io
       Destination: /workspaces/rustpython.github.io/_site
 Incremental build: disabled. Enable with --incremental
      Generating... 
       Jekyll Feed: Generating feed for posts
                    done in 10.022 seconds.
 Auto-regeneration: enabled for '/workspaces/rustpython.github.io'
                    ------------------------------------------------
      Jekyll 4.2.0   Please append `--trace` to the `serve` command 
                     for any additional information or backtrace. 
                    ------------------------------------------------
/usr/local/rvm/gems/ruby-3.2.2/gems/jekyll-4.2.0/lib/jekyll/commands/serve/servlet.rb:3:in `require': cannot load such file -- webrick (LoadError)
        from /usr/local/rvm/gems/ruby-3.2.2/gems/jekyll-4.2.0/lib/jekyll/commands/serve/servlet.rb:3:in `<top (required)>'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/jekyll-4.2.0/lib/jekyll/commands/serve.rb:179:in `require_relative'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/jekyll-4.2.0/lib/jekyll/commands/serve.rb:179:in `setup'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/jekyll-4.2.0/lib/jekyll/commands/serve.rb:100:in `process'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/jekyll-4.2.0/lib/jekyll/command.rb:91:in `block in process_with_graceful_fail'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/jekyll-4.2.0/lib/jekyll/command.rb:91:in `each'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/jekyll-4.2.0/lib/jekyll/command.rb:91:in `process_with_graceful_fail'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/jekyll-4.2.0/lib/jekyll/commands/serve.rb:86:in `block (2 levels) in init_with_program'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/mercenary-0.4.0/lib/mercenary/command.rb:221:in `block in execute'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/mercenary-0.4.0/lib/mercenary/command.rb:221:in `each'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/mercenary-0.4.0/lib/mercenary/command.rb:221:in `execute'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/mercenary-0.4.0/lib/mercenary/program.rb:44:in `go'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/mercenary-0.4.0/lib/mercenary.rb:21:in `program'
        from /usr/local/rvm/gems/ruby-3.2.2/gems/jekyll-4.2.0/exe/jekyll:15:in `<top (required)>'
        from /usr/local/rvm/gems/ruby-3.2.2/bin/jekyll:25:in `load'
        from /usr/local/rvm/gems/ruby-3.2.2/bin/jekyll:25:in `<main>'
        from /usr/local/rvm/gems/ruby-3.2.2/bin/ruby_executable_hooks:22:in `eval'
        from /usr/local/rvm/gems/ruby-3.2.2/bin/ruby_executable_hooks:22:in `<main>'
```
</details>

**Comment by olivierlemasle** [comment]
@homersimpsons I went here to fix the same benchmark issue (#66); I added a `.devcontainer` configuration instead :smile:: #67

Does this help you?

**Comment by homersimpsons** [comment]
> @homersimpsons I went here to fix the same benchmark issue (#66); I added a `.devcontainer` configuration instead 😄: #67
> 
> Does this help you?

Looks like I can run `jekyll` with your dev container, thank you!
